Contribute as SIG Docs reviewer blog post (#496)
kubernetes/contributor-site: content/en/blog/2024/contribute-as-sig-docs-reviewer.md

---
layout: blog

> title: "From Code to Quill: Embark on a Legendary Kubernetes Quest with SIG-Docs"

Reviewer: ```suggestion
title: "From Code to Quill: Embark on a Legendary Kubernetes Quest with SIG Docs"
```

Reviewer: No `-` in SIG names.

> date: 2024-04-28

Reviewer: We need a future date. How about the 16th of May; does that work?

Author: Works for me

slug: contribute-as-sig-docs-reviewer
author: "Ricardo Amaro (Acquia)"
---

You've probably come across the saying, "Contributing doesn't always mean writing lines of code"
whispered to the wind or glimpsed in the margins of a sacred open-source tome.
What is the truth behind this tale as old as time (or at least as old as Kubernetes itself)?

Imagine Sarah: an enthusiast of open-source whose keyboard had seen more action crafting
technical masterpieces for her projects than a knight's sword in battle. Sarah's had such
a smooth relationship with Git, that she could `git commit` and `git push` in her sleep,
and her etiquette in pull-requests was so polished, it could make a silver spoon look tarnished.

But here's the twist: when it came to the mysterious _art-of-coding_, Sarah felt more
like a fish out of water — or more accurately, like trying to navigate a Kubernetes cluster
with a compass and a star map. The very thought of diving into the codebase made her feel like

> she was trying to read ancient hieroglyphs without a Rosetta Stone.

Reviewer: ```suggestion
she was trying to read ancient hieroglyphs without a Rosetta stone.
```
Rosetta is a place (and, actually, so is Stone, but not in this sense).

Author: This is referring to https://en.wikipedia.org/wiki/Rosetta_Stone

Reviewer: I looked and most people write it with a capital S. So, the original is fine here. Sorry.

Author: All good!

Did Sarah let this stop her? Not at all. She discovered a path as impactful as a well-placed
line of code: joining the Special Interest Group (SIG) Docs team. It was there that Sarah found her true calling,
helping transform the dense field of Kubernetes documentation into a well-tended garden of knowledge.

Sarah's odyssey from a mere mortal Kubernetes user to a demigod of documentation didn't
just prove that contributions of knowledge and expertise are as valuable as code
contributions—it also showed that you don't need to be a coding wizard to make a significant
impact. Her journey is a beacon of hope for all who fear the merge conflict, a testament
to the power of the written word in the realm of open-source.

So, let Sarah's story inspire you. No matter if you're a code ninja or a prose maestro,
there's a place for you in the open-source pantheon. And who knows? Maybe one day,
your documentation will be the guiding light for a lost soul in the Kubernetes sea,
making you the unsung hero of their open-source odyssey.

> ## Why Become a SIG Docs Reviewer?

Reviewer: Headings in title case please:
```suggestion
## Why become a SIG Docs reviewer?
```

I think we're actually asking for people to join the _blog team_ though?

The Kubernetes project depends substantially on the SIG Docs team to ensure that documentation
is accurate, up to date, and easily accessible. By becoming a reviewer, you can help users
and contributors navigate and understand Kubernetes more effectively. In addition, reviewing
documentation provides unique opportunities for:

- **Expand your Kubernetes knowledge**: Engage deeply with new features and functionalities
  by reviewing their documentation.
- **Improve your technical writing skills**: Develop an eye for detail and clarity in technical
  writing.
- **Strengthen the Kubernetes community**: Help maintain the high quality of Kubernetes
  documentation, supporting both new and experienced users.
- **Build your network**: Expanding your professional network, and getting together with
  contributors from around the globe.

> ## Who Are We Looking For?

Reviewer: ```suggestion
## Who are we looking for?
```

We're seeking open-source enthusiasts with:

- Experience with Git and GitHub, comfortable with the process of reviewing pull requests
  and providing constructive feedback.
- Familiarity with Markdown and documentation frameworks (Hugo experience is a plus but
  not required).
- A passion for making complex technical concepts understandable and accessible.

Prior experience in technical writing or documentation review in any open-source project
will be beneficial but not mandatory.

> ## How to Get Started

Reviewer: ```suggestion
## How to get started
```

Becoming a SIG Docs reviewer is a journey that starts with familiarizing yourself with the

> Kubernetes documentation contribution process. Here are the steps to get you started:

Reviewer: I would talk about the _contributor ladder_ here. But, you might not be sure what this is. Ask in #sig-docs on Kubernetes Slack, or look at https://github.com/cncf/project-template/blob/main/CONTRIBUTOR_LADDER.md

   [#sig-docs channel](https://kubernetes.slack.com/messages/sig-docs). It's a great place
   to ask questions, find mentorship, and get to know the community.

1. **Start Reviewing Pull Requests**: Look for open pull requests labeled with
   `good first issue` or `help wanted`. These are great for beginners. Leave constructive
   feedback and suggestions. Familiarize yourself with the
   [content guide](/docs/contribute/style/content-guide/) and the 
   [style guide](/docs/contribute/style/style-guide/) to provide more effectively 
   [reviewing Pull Requests](/docs/contribute/review/reviewing-prs/).

1. **Attend SIG Docs Meetings**: Participate in the 
   [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs). These
   meetings are an excellent opportunity to meet fellow contributors, discuss documentation
   improvements, and volunteer for reviewing tasks.

1. **Shadow an Experienced Reviewer**: If possible, shadow an 
   [experienced reviewer](/docs/contribute/participate/roles-and-responsibilities/) 
   to learn best practices and tips for reviewing efficiently and effectively.

> ## What's Next?

Reviewer: ```suggestion
## What's next?
```

After consistently contributing to documentation reviews and demonstrating your understanding
of Kubernetes documentation standards, you can express your interest in becoming an official
SIG Docs reviewer. Engage with the SIG Docs chairs or leads on Slack or during SIG Docs

> meetings to discuss the next steps.

Reviewer: If we can, let's steer people to engage in the public channels, not via direct messages (those don't scale so well).

> ## See YOU in the Docs!

Reviewer: ```suggestion
## See _you_ in the docs!
```

_or_

```suggestion
## See _you_ in the community!
```

---
layout: blog
title: "From Code to Quill: Embark on a Legendary Kubernetes Quest with SIG Docs"
date: 2024-05-16
slug: contribute-as-sig-docs-reviewer
author: "Ricardo Amaro (Acquia)"
---

You've likely heard the adage, "Contributing isn't just about writing code",
whispered in forums or seen etched into the digital walls of open source communities.
But what depth of truth lies within this ancient wisdom, especially in the vast,
evolving world of Kubernetes?

Today, contributing to open source extends far beyond the realm of coding.
It's an inclusive journey that welcomes the diverse skills of all who wish
to be part of it. Whether you're a seasoned developer or someone whose strengths
lie outside traditional programming, your contributions hold immense value.

Imagine yourself embarking on a journey that leverages your expertise and love

> for free software to strengthen your capabilities and strengthen the very foundation

Reviewer 2: Consider replacing the first `strengthen` with say `bolster` or `enhance`

This is your call to adventure, an invitation to expand beyond the familiar territories
of code and into the rich, uncharted domains of documentation and collaboration,
empowering a global community of users and contributors.

## Why become a SIG Docs reviewer?

The Kubernetes project depends substantially on the SIG Docs to ensure that documentation
is accurate, up to date, and easily accessible. By becoming a reviewer, you can help users
and contributors navigate and understand Kubernetes more effectively. In addition, reviewing
documentation provides unique opportunities for:

- **Expanding your Kubernetes knowledge**: Engage deeply with new features and functionalities
  by reviewing their documentation.
- **Improving your technical writing skills**: Develop an eye for detail and clarity in technical
  writing.
- **Strengthening the Kubernetes community**: Help maintain the high quality of Kubernetes
  documentation, supporting both new and experienced users.
- **Building your network**: Expanding your professional network, and getting together with
  contributors from around the globe.

## Who are we looking for?

We're seeking open source enthusiasts with:

> - Experience with Git and GitHub, comfortable with the process of reviewing pull requests

Reviewer: ```suggestion
- Experience of Git and GitHub, comfortable with the process of reviewing pull requests
```

  and providing constructive feedback.
- Familiarity with Markdown and documentation frameworks (Hugo experience is a plus but
  not required).
- A passion for making complex technical concepts understandable and accessible.

> Prior experience in technical writing or documentation review in any open source project
> will be beneficial but not mandatory.

Reviewer: ```suggestion
Prior experience in technical writing or documentation review in any open source project
will be beneficial but not mandatory. We do need you to be very comfortable reading and writing
the language you'll be reviewing for (English is our main language).
```
You could mention what level of Kubernetes experience we look for: **not much**. If you have it, that's great;
if you don't yet know Kubernetes / containers, that means you bring a useful perspective that helps us check if documentation makes sense to a relative beginner.

## How to get started

Becoming a SIG Docs reviewer is a journey that starts with familiarizing yourself with the
Kubernetes documentation contribution process. Here are the steps to get you started:

1. **Familiarize Yourself with SIG Docs**: Start by reading the

>    [SIG Docs contributor guide](/docs/contribute/) to understand

Reviewer 3: ```suggestion
   [SIG Docs contributor guide](https://kubernetes.io/docs/contribute/) to understand
```

   how documentation contributions are made.

1. **Join the Kubernetes Slack**: Connect with the SIG Docs community on the

>    [#sig-docs channel](https://kubernetes.slack.com/messages/sig-docs). It's a great place

Reviewer: How about:
```suggestion
   [`#sig-docs` channel](https://kubernetes.slack.com/messages/C1J0BPD2M) (visit https://slack.k8s.io for
   an invitation, if you need one). It's a great place
```

> 1. **Join the Kubernetes Slack**: Connect with the SIG Docs community on the
>    [#sig-docs channel](https://kubernetes.slack.com/messages/sig-docs). It's a great place
>    to ask questions, find mentorship, and get to know the community.

Reviewer: How about:
```suggestion
   [`#sig-docs` channel](https://kubernetes.slack.com/messages/C1J0BPD2M) (visit https://slack.k8s.io for
   an invitation, if you need one). It's a great place
```
Permalinks are better in blog articles.

1. **Start Reviewing Pull Requests**: Look for open pull requests labeled with
   `good first issue` or `help wanted`. These are great for beginners. Leave constructive
   feedback and suggestions. Familiarize yourself with the

>    [content guide](/docs/contribute/style/content-guide/) and the 

Reviewer 3: ```suggestion
   [content guide](https://kubernetes.io/docs/contribute/style/content-guide/) and the 
```

>    [style guide](/docs/contribute/style/style-guide/) to provide more effectively 

Reviewer 3: ```suggestion
   [style guide](https://kubernetes.io/docs/contribute/style/style-guide/) to provide more effectively 
```

>    [reviewing Pull Requests](/docs/contribute/review/reviewing-prs/).

Reviewer 3: ```suggestion
   [reviewing Pull Requests](https://kubernetes.io/docs/contribute/review/reviewing-prs/).
```

1. **Attend SIG Docs Meetings**: Participate in the 
   [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs). These
   meetings are an excellent opportunity to meet fellow contributors, discuss documentation
   improvements, and volunteer for reviewing tasks.

1. **Shadow an Experienced Reviewer**: To better understand the review process, sign up for our official

>    [PR Wrangling shadow program](/docs/contribute/participate/pr-wranglers/#pr-wrangler-shadow-program)

Reviewer 3: ```suggestion
   [PR Wrangling shadow program](https://kubernetes.io/docs/contribute/participate/pr-wranglers/#pr-wrangler-shadow-program)
```

>    where you will be able to shadow an [experienced reviewer](/docs/contribute/participate/roles-and-responsibilities/) 

Reviewer 3: ```suggestion
   where you will be able to shadow an [experienced reviewer](https://kubernetes.io/docs/contribute/participate/roles-and-responsibilities/) 
```

   and learn best practices and tips for reviewing efficiently and effectively.

1. **Contributor Ladder**: Familiarize yourself with the
   [CNCF contributor ladder](https://github.com/cncf/project-template/blob/main/CONTRIBUTOR_LADDER.md),
   which guides your progression from newcomer to advanced contributor roles in the Kubernetes community.

## What's next?

After consistently contributing to documentation reviews and demonstrating your understanding
of Kubernetes documentation standards, you can express your interest in becoming an official
SIG Docs reviewer. Engage with the SIG Docs chairs or leads on the

> [public Slack channel](https://kubernetes.slack.com/channels/sig-docs)

Reviewer: ```suggestion
[public Slack channel](https://kubernetes.slack.com/channels/C1J0BPD2M)
```
Permalinks are better in blog articles.

---
layout: blog
title: "From Code to Quill: Embark on a Legendary Kubernetes Quest with SIG Docs"
date: 2024-05-16
slug: contribute-as-sig-docs-reviewer
author: "Ricardo Amaro (Acquia)"
---

You've probably come across the saying, "Contributing doesn't always mean writing lines of code"
whispered to the wind or glimpsed in the margins of a sacred open-source tome.
What is the truth behind this tale as old as time (or at least as old as Kubernetes itself)?

Imagine Sarah: an enthusiast of open-source whose keyboard had seen more action crafting
technical masterpieces for her projects than a knight's sword in battle. Sarah's had such
a smooth relationship with Git, that she could `git commit` and `git push` in her sleep,
and her etiquette in pull-requests was so polished, it could make a silver spoon look tarnished.

> But here's the twist: when it came to the mysterious _art-of-coding_, Sarah felt more

Reviewer 4: Has Sarah signed off on this paragraph (and narrative in general)?  I'm worried that it overemphasizes her lack of programming skills.  If she learns to program next year, is this going to be embarassing?

Also, consider `_art of programming_` instead.  "coding" feels weird in this context.

Author: Sarah is a fictional character. We might write something more in the future, you are right. But up to now, this is the story to be unveiled further in the next chapters.

Author: Updating to `_art of programming_`. Thank you!

Reviewer 4: I am opposed to writing this story with a fictional character, particularly one that's female.

Author: Removed the fictional character with a more inclusive "achievement journey" aimed directly at the reader.

like a fish out of water — or more accurately, like trying to navigate a Kubernetes cluster
with a compass and a star map. The very thought of diving into the codebase made her feel like
she was trying to read ancient hieroglyphs without a Rosetta Stone.

Did Sarah let this stop her? Not at all. She discovered a path as impactful as a well-placed

> line of code: joining the Special Interest Group (SIG) Docs team. It was there that Sarah found her true calling,

Reviewer 4: Do we need to spell out SIG for our audience?

Author: The blog post is also addressed to people outside the community that might not be familiarized with the term.

helping transform the dense field of Kubernetes documentation into a well-tended garden of knowledge.

Sarah's odyssey from a mere mortal Kubernetes user to a demigod of documentation didn't
just prove that contributions of knowledge and expertise are as valuable as code
contributions—it also showed that you don't need to be a coding wizard to make a significant
impact. Her journey is a beacon of hope for all who fear the merge conflict, a testament

> to the power of the written word in the realm of open-source.

Reviewer 4: So ... we don't get the actual story?  It seems to be missing here.  I feel like I've read the prologue and then chapters 2 through 15 are torn out of the book.

If you're going to write a pulp adventure story about a docs reviewer, you need to finish the story.

Author: This is written in Cordel literature form and is meant to incentivize people to contribute. The journey is just starting for her and for the newcomers. ☺️

So, let Sarah's story inspire you. No matter if you're a code ninja or a prose maestro,
there's a place for you in the open-source pantheon. And who knows? Maybe one day,
your documentation will be the guiding light for a lost soul in the Kubernetes sea,
making you the unsung hero of their open-source odyssey.

## Why become a SIG Docs reviewer?

The Kubernetes project depends substantially on the SIG Docs team to ensure that documentation
is accurate, up to date, and easily accessible. By becoming a reviewer, you can help users
and contributors navigate and understand Kubernetes more effectively. In addition, reviewing
documentation provides unique opportunities for:

> - **Expand your Kubernetes knowledge**: Engage deeply with new features and functionalities

Reviewer 2: Since you've written `reviewing`, I think `Expand` should `Expanding` and so on.

Just my thoughts.

Author: Thanks!

  by reviewing their documentation.
- **Improve your technical writing skills**: Develop an eye for detail and clarity in technical
  writing.

> - **Strengthen the Kubernetes community**: Help maintain the high quality of Kubernetes

Reviewer 2: `Strengthening`

  documentation, supporting both new and experienced users.

> - **Build your network**: Expanding your professional network, and getting together with

Reviewer 2: `Building`
